warehouse/email/__init__.py

```diff
@@ -33,12 +35,28 @@ def _compute_recipient(user, email):
 
 
 @tasks.task(bind=True, ignore_result=True, acks_late=True)
-def send_email(task, request, recipient, msg):
+def send_email(
+    task, request, recipient, msg, sending_user_id, ip_address, redact_ip=True
+):
```

Comment:
I think I'd prefer:

```suggestion
def send_email(task, request, recipient, msg, success_event):
```

and below we just do:

```python
user_service = request.find_service(IUserService, context=None)
user_service.record_event(success_event)
```

Reply:
I was able to reduce the number of parameters we pass into `send_email` by passing a dict but I can't call the `record_event` function exactly as you're expecting because some of the parameters for `record_event` such as the `from_address` come from inside the `send_email` task. Does that work?

Reply:
I think we can determine `from_address` in the view as well.